### PR TITLE
test: Tests logging on some LoggingStorage's operations

### LINKED ISSUE #345: Test that operations are properly logged in LoggingStorage. We are currently testing that...
The puzzle `186-fbb1def2` from #186 has to be resolved: 

https://github.com/artipie/asto/blob/7e07dd52496ceb2dae39a264a37336e04902b2a3/src/test/java/com/artipie/asto/LoggingStorageTest.java#L19-L22

The puzzle was created by @baudoliver7 on 27-Oct-21. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by baudoliver7**
Closes #345

Test logging on methods `save`, `exists` and `list`

### CODE REVIEW

**Comment by baudoliver7** [comment]
@artipie/maintainers, needs a review here :)

File: src/test/java/com/artipie/asto/LoggingStorageTest.java [hunk 1]
```diff
@@ -102,4 +115,45 @@ void saveOverwrites() {
...
+
+    @Test
+    void logsWhenListingKeys() {
+        this.memsto.save(
+            new Key.From("one/two"),
+            new Content.From("data1".getBytes(StandardCharsets.UTF_8))
+        ).join();
+        this.memsto.save(
+            new Key.From("one/three"),
+            new Content.From("data2".getBytes(StandardCharsets.UTF_8))
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/asto/LoggingStorageTest.java:127]: as content of the storage item does not matter here, `Content.EMPTY` can be used

File: src/test/java/com/artipie/asto/LoggingStorageTest.java [hunk 2]
```diff
@@ -102,4 +115,45 @@ void saveOverwrites() {
...
+            this.writer.toString(),
+            new IsEqual<>("List 'one': 2")
+        );
+    }
+
+    @Test
+    void logsWhenCheckingExistence() {
+        final byte[] data = "0101".getBytes(StandardCharsets.UTF_8);
+        final Key key = new Key.From("binary-file");
+        this.memsto.save(key, new Content.From(data)).join();
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/asto/LoggingStorageTest.java:140]: same here

File: src/test/java/com/artipie/asto/LoggingStorageTest.java [hunk 3]
```diff
@@ -102,4 +115,45 @@ void saveOverwrites() {
...
+
+    @Test
+    void logsWhenSaving() {
+        final byte[] data = "content".getBytes(StandardCharsets.UTF_8);
+        final Key key = new Key.From("key");
+        final LoggingStorage logsto = new LoggingStorage(this.memsto);
+        logsto.save(key, new Content.From(data)).join();
+        MatcherAssert.assertThat(
+            this.writer.toString(),
+            new IsEqual<>("Save 'key': 7")
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/asto/LoggingStorageTest.java:156]: I'd suggest use `data.length` in the assertion instead of stub string

**Review by olenagerasimova — APPROVED** [review]
@baudoliver7 thanks, added some comments

File: src/test/java/com/artipie/asto/LoggingStorageTest.java (hunk 1 above)
Comment (baudoliver7) [reply src/test/java/com/artipie/asto/LoggingStorageTest.java:127]: @olenagerasimova You are right !

**Review by olenagerasimova — DISMISSED** [review]
@baudoliver7 thanks
@g4s8 please, take a look/merge

**Comment by baudoliver7** [comment]
@g4s8 please, need check/merge here

File: src/main/java/com/artipie/asto/LoggingStorage.java [hunk 4]
```diff
@@ -72,7 +72,7 @@ public LoggingStorage(final Level level, final Storage storage) {
     public CompletableFuture<Void> save(final Key key, final Content content) {
         return this.storage.save(key, content).thenApply(
             result -> {
-                this.log("Save '%s': %s", key.string(), content.size());
+                this.log("Save '%s': %s", key.string(), content.size().get());
```
Comment (g4s8) [inline src/main/java/com/artipie/asto/LoggingStorage.java:75]: What if the size is empty here?

**Review by g4s8 — CHANGES_REQUESTED** [review]
@baudoliver7 looks good, please check two comments

File: pom.xml [hunk 5]
```diff
@@ -93,6 +93,11 @@ SOFTWARE.
       <artifactId>jcabi-log</artifactId>
       <version>0.20.1</version>
     </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-log4j12</artifactId>
+      <version>2.0.0-alpha5</version>
+    </dependency>
```
[the remark this answers is not in the document]
Comment (baudoliver7) [reply pom.xml:100]: @g4s8 Yes, it is. Let's me add a test scope :)

**Review by g4s8 — APPROVED** [review]
@baudoliver7 thanks

**Pull request merged**